snarky_monad_lib/dune

```diff
@@ -0,0 +1,9 @@
+(library
+ (name snarky_monad_lib)
+ (public_name snarky.monad_lib)
+ (library_flags -linkall)
```

Comment:
I would check if it is necessary with the recent dune versions. I know it's everywhere in the codebase, but not sure it is required.